pkg/client/interfaces_test.go

```diff
@@ -32,6 +34,72 @@ func TestConfigValidation(t *testing.T) {
 		valid         bool
 		expectedError string
 	}{
+		{
+			desc: "gen config with valid yaml value is valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yaml": "test: \"valid\"",
+						},
+					},
+				},
+			},
+			valid: true,
+		},
+		{
+			desc: "gen config with valid yml value is valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yml": "test: \"valid\"",
+						},
+					},
+				},
+			},
+			valid: true,
+		},
+		{
+			desc: "gen config with invalid yaml value is not valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yaml": "test: \"in\"valid\"",
+						},
+					},
+				},
+			},
+			valid:         false,
+			expectedError: "failed to parse value of key test.yaml in ConfigMap for plugin test: error converting YAML to JSON: yaml: did not find expected key",
+		},
+		{
+			desc: "gen config with invalid yml value is not valid",
+			config: &GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{
+						SonobuoyConfig: manifest.SonobuoyConfig{
+							PluginName: "test",
+						},
+						ConfigMap: map[string]string{
+							"test.yml": "test: \"in\"valid\"",
+						},
+					},
+				},
+			},
+			valid:         false,
+			expectedError: "failed to parse value of key test.yml in ConfigMap for plugin test: error converting YAML to JSON: yaml: did not find expected key",
+		},
```

Comment:
Current test cases are good; lets add just a few more to cover a few different edge cases. 
 - A plugin with 2 configmaps, one valid one invalid to ensure we never introduce an early exit from the checks
 - A case where its not valid yaml but its not named "*.yaml" so we ensure we only test the intended files